src/writer/update-proposal.ts

```diff
@@ -0,0 +1,186 @@
+import isEqual from 'lodash/isEqual';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { getAddress } from '@ethersproject/address';
+import { jsonParse } from '../helpers/utils';
+import db from '../helpers/mysql';
+import { getSpace, getProposal } from '../helpers/actions';
+import log from '../helpers/log';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import {
+  flaggedAddresses,
+  flaggedProposalTitleKeywords,
+  flaggedProposalBodyKeywords
+} from '../helpers/moderation';
+
+const scoreAPIUrl = process.env.SCORE_API_URL || 'https://score.snapshot.org';
+
+export function isChoicesValid({ type, choices }): boolean {
```

Comment:
What happens if some choices are being removed, while there are already existing votes with the removed choices ?

Reply:
But what happens to existing votes with `==` which do no longer exist in the voting choices, but have been submitted ?

Reply:
> What happens if some choices are being removed, while there are already existing votes with the removed choices ?

You can only update proposal before it start

Reply:
Did not see any check related to that